ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/DefaultSynchronizer.java

```diff
@@ -205,6 +215,9 @@ public TrailingPeerRequirements calculateTrailingPeerRequirements() {
   public CompletableFuture<Void> start() {
     if (running.compareAndSet(false, true)) {
       LOG.info("Starting synchronizer.");
+
+      overallSyncTimingContext = overallSyncTimer.startTimer();
```

Comment:
Not sure the prometheus metrics are designed to time such a long task, could have implications on memory for example.

Was thinking a metric to record the events "sync started" and "sync finished" would be better, then can derive the duration after the fact, e.g. in the dashboard panel. Also, makes the metrics more flexible in case we want to combine the various events, e.g. for different sync phases.

Reply:
Am starting to wonder if metrics is the right way to do this - maybe parsing the logs with Loki is a better way to go? Might be a bit more effort to set up though. Can you see any long term advantage of metrics over logs?

Reply:
In my opinion it is better to use code (Metrics) than logs, because people might change logs when they are not aware that they are used in Grafana.